src/icons.ts

```diff
@@ -0,0 +1,156 @@
+/*-------------------------------------------------------------------------------------------------
+ *  Copyright (C) 2023 Intel Corporation. All rights reserved.
+ *  Licensed under the Apache License 2.0. See LICENSE in the project root for license information.
+ *  SPDX-License-Identifier: Apache-2.0
+ *-----------------------------------------------------------------------------------------------*/
+
+import * as H from "./helper";
+import fs from "graceful-fs";
+import path from "path";
+import { Project } from "./project";
+import multiparty from "multiparty";
+import errorCode from "./error_code";
+
+const log = H.getLogger("icons");
+
+const ProjectIconFolder = ".webinizer/icons";
+
+export interface Icons {
+  name: string;
+  isUploaded: boolean;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function handleUploadIcon(root: string, req: any, res: any) {
+  const multiParty = new multiparty.Form(req);
+  multiParty.parse(req, async (err, fields, files) => {
+    try {
+      if (err) throw err;
+      const fileName = path.parse(String(files.file[0].originalFilename)).name;
+      const fileSize = Number(files.file[0].size);
+
+      checkUploadIconType(files.file[0].path);
+      checkIfFileSize(fileSize);
+
+      const targetFilePath = constructIconPath(root, fileName);
+      fs.renameSync(String(files.file[0].path), targetFilePath);
+
+      // update project config
+      const proj = new Project(root);
+      const imgName = path.basename(targetFilePath);
```

Comment:
The uploaded icon image is saved without extension names as `20231023T011916350Z-test`.

Maybe use `path.parse(String(files.file[0].originalFilename)).ext` to get the extension and construct the `targetFilePath` together.